src/screenshotdialog.c

```diff
@@ -30,10 +47,281 @@ static guint signals[LAST_SIGNAL];
 
 G_DEFINE_TYPE (ScreenshotDialog, screenshot_dialog, GTK_TYPE_WINDOW)
 
+static void
+update_border (ScreenshotDialog *dialog,
+               const char *kind)
+{
+  GAction *border;
+  GAction *pointer;
+
+  border = g_action_map_lookup_action (G_ACTION_MAP (dialog->actions), "border");
+  pointer = g_action_map_lookup_action (G_ACTION_MAP (dialog->actions), "pointer");
+
+  if (strcmp (kind, "screen") == 0)
+    {
+      g_simple_action_set_state (G_SIMPLE_ACTION (border), g_variant_new_boolean (FALSE));
+      g_simple_action_set_enabled (G_SIMPLE_ACTION (border), FALSE);
```

Comment:
While testing this with PortalTest, I noticed that the "Include the window border" is not disabled the first time the options are shown in the dialog. Either initializing this to disabled in the .ui file or calling `update_border(dialog, "screen")` on initialization should fix that.

Reply:
I'd say calling `update_border()` sounds probably better, though...

Reply:
As long as the setting is consistent with what's actually done while taking the screenshot, I'm not strongly opposed to it. Thing is, IIRC it was not consistent on the first time when I tested it and the screenshot was taken in a way that it was not honouring what the UI  showed, but perhaps I got it wrong.

I can't test it right now, but it would be worth double-checking just in case.